minichef/contracts/Strategy.sol

```diff
@@ -0,0 +1,424 @@
+        assembly {
+            // EIP-1167 bytecode
+            let clone_code := mload(0x40)
+            mstore(
+                clone_code,
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newStrategy := create(0, clone_code, 0x37)
+        }
+
+        Strategy(newStrategy).initialize(
+            _vault,
+            _strategist,
+            _rewards,
+            _keeper,
+            _pid,
+            _name
+        );
+
+        emit Cloned(newStrategy);
+    }
+
+    // These functions are useful for setting parameters of the strategy that may need to be adjusted.
+    // Set optimal token to sell harvested funds for depositing to Curve.
+    // Default is DAI, but can be set to USDC, USDT as needed by strategist or governance.
+    function setOptimal(uint256 _optimal) external onlyAuthorized {
+        if (_optimal == 0) {
+            targetToken = address(dai);
+            optimal = 0;
+        } else if (_optimal == 1) {
+            targetToken = address(usdc);
+            optimal = 1;
+        } else if (_optimal == 2) {
+            targetToken = address(usdt);
+            optimal = 2;
+        } else if (_optimal == 3) {
+            targetToken = address(nusd);
+            optimal = 3;
+        }
+    }
+
+    ///@notice Only do this if absolutely necessary; as assets will be withdrawn but rewards won't be claimed.
+    function emergencyWithdraw() external onlyEmergencyAuthorized {
+        chef.emergencyWithdraw(pid, address(this));
+    }
+
+    
+    
+    function protectedTokens() internal view override returns (address[] memory) {
+    }
+
+
+
+
+    function liquidatePosition(uint256 _amountNeeded)
+        internal
+        override
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        uint256 totalAssets = want.balanceOf(address(this));
+        if (_amountNeeded > totalAssets) {
+            uint256 amountToFree = _amountNeeded.sub(totalAssets);
+
+            uint256 deposited = balanceOfStaked();
+            if (deposited < amountToFree) {
+                amountToFree = deposited;
+            }
+            if (deposited > 0) {
+                chef.withdraw(pid, amountToFree, address(this));
+            }
+
+            _liquidatedAmount = want.balanceOf(address(this));
+        } else {
+            _liquidatedAmount = _amountNeeded;
+        }
+    }
+
+    // withdraw all funds
+    function liquidateAllPositions() internal override returns (uint256) {
+        uint256 amount = balanceOfStaked();
+        if (amount > 0) {
+            chef.withdrawAndHarvest(
+                pid, 
+                amount, 
+                address(this)
+            );
+            return balanceOfWant();
```

Comment:
Comparing to the original strategy, I think we should move this return outside of the if: https://github.com/dudesahn/Strategy0xDAOStaker/blob/main/contracts/Strategy0xDAOStaker.sol#L341